### PR TITLE
rgw/lc: remove_bucket_config() doesn't update xattrs on bucket delete

### PR DESCRIPTION
**Pull request by cbodley**
on bucket deletion, we're going to delete the bucket instance metadata anyway so there's no reason for `RGWLC::remove_bucket_config()` to write an update that removes `RGW_ATTR_LC` first

and if the bucket doesn't have an `RGW_ATTR_LC`, we shouldn't need to remove it from the lc list either

this relates to https://tracker.ceph.com/issues/62411, but i don't think it really addresses the underlying issue

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: src/rgw/driver/rados/rgw_sal_rados.cc [hunk 1]
```diff
@@ -452,7 +452,7 @@ int RadosBucket::remove_bucket(const DoutPrefixProvider* dpp,
 
   // remove lifecycle config, if any (XXX note could be made generic)
   (void) store->getRados()->get_lc()->remove_bucket_config(
-    this, get_attrs());
+    this, get_attrs(), false);
```
Comment (mattbenjamin) [inline src/rgw/driver/rados/rgw_sal_rados.cc]: what does the false do?  maybe an inline comment there?

**Review by mattbenjamin — APPROVED** [review]

File: src/rgw/driver/rados/rgw_sal_rados.cc (hunk 1 above)
Comment (cbodley) [reply src/rgw/driver/rados/rgw_sal_rados.cc]: thanks, i added a constexpr variable to name that argument, and added a comment saying why

**Comment by mattbenjamin** [comment]
nice

**Comment by cbodley** [comment]
passed qa in https://pulpito.ceph.com/cbodley-2023-09-29_14:54:02-rgw-wip-cbodley-testing-distro-default-smithi/

**Pull request merged**